azext_iot/tests/dps/device_registration/test_iot_device_registration_individual_int.py

```diff
@@ -0,0 +1,541 @@
+                    self.check("registrationState.registrationId", CERT_NAME),
+                    self.check("registrationState.substatus", "initialAssignment"),
+                    self.check("status", "assigned"),
+                ],
+            )
+            self.check_hub_device(device_id, "selfSigned", thumbprint=secondary_thumprint)
+
+            self.cmd(
+                self.set_cmd_auth_type(
+                    "iot dps enrollment delete --enrollment-id {} -g {} --dps-name {}".format(
+                        CERT_NAME,
+                        self.entity_rg,
+                        self.entity_dps_name,
+                    ),
+                    auth_type=auth_phase
+                ),
+            )
+
+    def test_dps_device_registration_unlinked_hub(self):
+        # Unlink hub
+        self.cmd(
+            "iot dps linked-hub delete --dps-name {} -g {} --linked-hub {}".format(
+                self.entity_dps_name,
+                self.entity_rg,
+                self.entity_hub_name
+            )
+        )
+
+        attestation_type = AttestationType.symmetricKey.value
+        for auth_phase in DATAPLANE_AUTH_TYPES:
+            enrollment_id = self.generate_enrollment_names()[0]
+
+            self.cmd(
+                self.set_cmd_auth_type(
+                    "iot dps enrollment create --enrollment-id {} --attestation-type {}"
+                    " -g {} --dps-name {}".format(
+                        enrollment_id,
+                        attestation_type,
+                        self.entity_rg,
+                        self.entity_dps_name,
+                    ),
+                    auth_type=auth_phase
+                ),
+            )
+
+            # registration throws error
+            self.cmd(
+                self.set_cmd_auth_type(
+                    "iot device registration create --dps-name {} -g {} --registration-id {}".format(
+                        self.entity_dps_name, self.entity_rg, enrollment_id
+                    ),
+                    auth_type=auth_phase
+                ),
+                expect_failure=True
+            )
+
+            # Can see registration
+            self.cmd(
+                self.set_cmd_auth_type(
+                    "iot dps enrollment registration show --dps-name {} -g {} --enrollment-id {}".format(
+                        self.entity_dps_name, self.entity_rg, enrollment_id
+                    ),
+                    auth_type=auth_phase
+                ),
+                checks=[
+                    self.exists("etag"),
+                    self.exists("lastUpdatedDateTimeUtc"),
+                    self.check("registrationId", enrollment_id),
+                    self.check("status", "failed"),
+                ],
+            )
+
+    def test_dps_device_registration_disabled_enrollment(self):
+        attestation_type = AttestationType.symmetricKey.value
+        for auth_phase in DATAPLANE_AUTH_TYPES:
```

Comment:
For enrollment[group] all the DATAPLANE_AUTH_TYPES should be tested in the respective enrollment[group] tests. Device registration tests can iterate the auth types for the new functionality being presented in the PR (like `az iot device registration create`) but do not need to iterate auth types for other elsewhere tested functionality.